core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/SequencingParameters.java

```diff
@@ -2,39 +2,50 @@
 
 import java.util.Date;
 
+import javax.xml.xpath.XPathExpressionException;
+
+import org.w3c.dom.Document;
+
 import com.eaglegenomics.simlims.core.User;
 
-public interface SequencingParameters {
+public interface SequencingParameters extends Comparable<SequencingParameters> {
+
+  public User getCreatedBy();
+
+  public Date getCreationDate();
+
   public Long getId();
 
-  public void setId(Long id);
+  public Date getLastUpdated();
 
   public String getName();
 
-  public void setName(String string);
+  public Platform getPlatform();
 
   /**
    * Gets the platform ID. This exists due to platforms not being in Hibernate.
    */
   public Long getPlatformId();
 
-  public Platform getPlatform();
+  public User getUpdatedBy();
 
-  public void setPlatform(Platform platform);
+  String getXPath();
 
-  public User getCreatedBy();
+  boolean matches(Document document) throws XPathExpressionException;
 
   public void setCreatedBy(User createdby);
 
-  public Date getCreationDate();
-
   public void setCreationDate(Date creation);
 
-  public User getUpdatedBy();
+  public void setId(Long id);
 
-  public void setUpdatedBy(User updatedBy);
+  public void setLastUpdated(Date lastUpdated);
 
-  public Date getLastUpdated();
+  public void setName(String string);
 
-  public void setLastUpdated(Date lastUpdated);
+  public void setPlatform(Platform platform);
+
+  public void setUpdatedBy(User updatedBy);
+
+  void setXPath(String xpath);
 }
```

Comment:
The re-alphabetizing of getters and setters makes this somewhat confusing, but it looks like everything is in (a new) order.